components/script/dom/element.rs

```diff
@@ -706,15 +705,17 @@ impl<'a> AttributeHandlers for JSRef<'a, Element> {
             .map(|x| Temporary::from_rooted(x.r()))
     }
 
-    fn get_attribute_by_name(self, name: &Atom) -> Option<Temporary<Attr>> {
+
+    // https://dom.spec.whatwg.org/#concept-element-attributes-get-by-name
+    fn get_attribute_by_name(self, name: DOMString) -> Option<Temporary<Attr>> {
+        let name = self.parsed_name(name);
         // FIXME(https://github.com/rust-lang/rust/issues/23338)
         let attrs = self.attrs.borrow();
         attrs.iter().map(|attr| attr.root())
-             .find(|a| a.r().name() == name)
+             .find(|a| a.r().Name() == name)
```

Comment:
I've changed here, because ```Name()``` returns ```DOMString``` and ```name()``` returns ```Atom``` type. I have ```DOMString``` from ```self.parsed_name(name)```. I'm not sure which version is faster (comparing Atom types or DOMString types), but this one seems cleaner.